Added tests in submitter and changed metadata_context

dcdarrell9 opened the pull request:
### What is the context of this PR?
Added tests for coverage and changed some code.

### How to review 
- Check to see if the tests do what they should.
- Is the code in metadata_context worth keeping? (validated elsewhere)
- Suggest improvements.

dcdarrell9 committed:
Added tests in submitter and changed metadata_context

jonnyshaw89 commented on tests/app/submitter/test_submitter.py:
```diff
@@ -78,3 +78,30 @@ def test_when_fail_to_publish_message_then_returns_false(self):
+    def test_get_submitter_settings_enabled(self):
+        settings.EQ_RABBITMQ_ENABLED = True
+        test = SubmitterFactory.get_submitter()
+        self.assertIsInstance(type(test), cls=object)
```
Should we also test that the correct object is returned?

dcdarrell9 replied:
Agree, I know the right object is returned because it prints it into the terminal but I don't know how to check it?

jonnyshaw89 replied:
Can you do something like?
`self.assertIsInstance(type(test), RabbitMQSubmitter)`

dcdarrell9 replied:
`IsInstance` doesn't work, but `assertEqual` works, is that right?

jonnyshaw89 replied:
If it works i guess so 👍

jonnyshaw89 commented on tests/app/submitter/test_submitter.py:
```diff
@@ -78,3 +78,30 @@ def test_when_fail_to_publish_message_then_returns_false(self):
+
+    def test_submitter_encrypt_message(self):
+        test = Submitter.encrypt_message(message={})
+        self.assertIsInstance(test, cls=str)
```
You should test that the messages is actually being encrypted rather than just writing a test to touch lines.

jonnyshaw89 commented on tests/app/submitter/test_submitter.py:
```diff
@@ -78,3 +78,30 @@ def test_when_fail_to_publish_message_then_returns_false(self):
+
+    def test_submitter_send_message(self):
+        test = Submitter.send_message(self, message={}, queue='test_queue')
+        self.assertEqual(test, False)
```
The name of the test implies that the result should be True?

dcdarrell9 replied:
I'm not sure about this one because it's an abstract method, do they get tested differently?

jonnyshaw89 replied:
Maybe test send_message with both the LogSubmitter and the RabbitMQSubmitter?
each will have a different action.

dcdarrell9 committed:
edited tests

dcdarrell9 commented on tests/app/submitter/test_submitter.py:
```diff
@@ -78,3 +78,31 @@ def test_when_fail_to_publish_message_then_returns_false(self):
+    def test_submitter_encrypt_message(self):
+        encrypt_message_test = Submitter.encrypt_message(message={'test': 'test'})
+        self.assertIsInstance(encrypt_message_test, cls=str)
+        self.assertNotEqual(encrypt_message_test, 'test')
```
From what I think I know... 'test' would get printed if it wasn't encrypted, am I right to say that?

jonnyshaw89 replied:
yes that is correct. however the test isn't. this is currently testing that the entire message content is 'test' rather than the messages containing the word test

Commit:
Change back to original

Wait for the change from Alex's settings change to carry on with this part of the PR, other change is fine to go through
